megamek/src/megamek/common/Entity.java

```diff
@@ -15565,5 +15567,88 @@ public int modifyPhysicalDamageForMeleeSpecialist() {
 
         return 1;
     }
-
+    
+    
+    //Getters and setters for sensor contacts and firing solutions. Currently only used in space combat
+    /**
+     * Retrieves the IDs of all entities that this entity has detected with sensors
+     * @return the contents of this entity's sensorContacts set
+     */
+    public Set<Integer> getSensorContacts() {
+        return sensorContacts;
+    }
+    
+    /**
+     * Checks the sensorContacts set for a specific target's ID number
+     * @param targetId the ID number of the target entity to check for
+     * @return true if the entity's sensorContacts set contains the passed-in target ID
+     */
+    public boolean hasSensorContactFor(int targetId) {
+        return sensorContacts.contains(targetId);
+    }
+    
+    /**
+     * Adds the specified target entity's ID to this entity's sensorContacts
+     * @param targetId the ID number of the target entity to add
+     */
+    public void addSensorContact(int targetId) {
+        sensorContacts.add(targetId);
+    }
+    
+    /**
+     * Removes the specified target entity's ID from this entity's sensorContacts
+     * @param targetId the ID number of the target entity to remove
+     */
+    public void removeSensorContact(int targetId) {
```

Comment:
Perhaps make this:
```java
public void removeSensorContacts(Collection<int> targetIds) {
    sensorContacts.removeAll(targetIds);
}
```
Then you can just pass `toRemove` directly to this, no need to check for `size >= 1`.

Reply:
Done.  Didn't recognize that I could do it this way.